scripts/skills/perks/perk_rf_hidden_pockets.nut

```diff
@@ -0,0 +1,54 @@
+this.perk_rf_hidden_pockets <- ::inherit("scripts/skills/skill", {
+	m = {},
+	function create()
+	{
+		this.m.ID = "perk.rf_hidden_pockets";
+		this.m.Name = ::Const.Strings.PerkName.RF_HiddenPockets;
+		this.m.Description = ::Const.Strings.PerkDescription.RF_HiddenPockets;
+		this.m.Icon = "ui/perks/rf_hidden_pockets.png";
+		this.m.Type = ::Const.SkillType.Perk;
+		this.m.Order = ::Const.SkillOrder.Last;
+		this.m.IsActive = false;
+		this.m.IsStacking = false;
+		this.m.IsHidden = false;
+	}
+
+	function onUpdate( _properties )
+	{
+		// This works exactly like how the vanilla bags and belts perk adds its slots. Both at the same time will therefor not have any effect
+		this.getContainer().getActor().getItems().setUnlockedBagSlots(4);
+	}
+
+	function onRemoved()
+	{
+		this.getContainer().removeByID("actives.rf_pocket_sand_skill");
+
+		// Recover items from the lost bag slots when this perk is removed. This is a copy of how vanilla does it
+		local items = this.getContainer().getActor().getItems();
+		local item;
+		item = items.getItemAtBagSlot(2);
+
+		if (item != null)
+		{
+			items.removeFromBag(item);
+			this.World.Assets.getStash().add(item);
+		}
+
+		item = items.getItemAtBagSlot(3);
+
+		if (item != null)
+		{
+			items.removeFromBag(item);
+			this.World.Assets.getStash().add(item);
+		}
+
+		// This line exists in bags and belts aswell. Let's hope that there happens another onUpdate over all skills right after.
+		// Otherwise this can cause problems with both Bags and Belts and Hidden Pockets were allocated, but only one was removed
```

Comment:
I'd suggest to please test this in-game and find out exactly what happens in this case. And if something unexpected happens then we can try to fix it.

Reply:
There is no way in vanilla or reforged to unlearn a single perk point so this can never occur. Even if there was, this would still be a rare edge case. 

I don't plan on testing for this rare edge case. If you feel better I can remove this comment

Reply:
No, please test it. It isn't an elaborate test:
- Add both perks via Taro's dev console.
- Fill the bags with items. 
- Then remove bags and belts. 
- Repeat the test but this time remove Trickster's Purses instead.

Reply:
I tested that out now.
The result is actually not suprising:
- If I'm in the character/inventory screen and refund a random perk (e.g. with that suggested potion) then nothing happens at first.
- Once I reopen the menu **or** equip/unequip any item to the brother who just got backs&pelts or tricksters purses unlearned then suddenly he unequips both items in the slot 3 and 4 of his bags. Even if he still has 4 slots afterwards. This is unexpected behavior from the `onRemoved` function of both perks, they don't check if the respective other perk is present

So as of right now this does not void any items, unless you unlearn such a perk with a full inventory, but that already happens in vanilla when using the oblivion potion. The observed behavior is a bit weird but does not break anything at the moment.

### Future

We should look into managing the unlocked bagslots in a more central location, currently that is done within the `item_container`.

Maybe we can use `bag_fatigue` skill to manage the unlocked bagslots.
Or we move the unlocked bagslots into the `character.nut` which means that it can now be manipulated during the update by changing the value in `_properties`

Reply:
Thank you for testing. Yes it is sufficient to update the comment so we know what to expect. We don't have to implement any fix for this behavior.